presto-orc/src/main/java/com/facebook/presto/orc/StreamDescriptor.java

```diff
@@ -75,12 +89,27 @@ public OrcDataSource getOrcDataSource()
         return nestedStreams;
     }
 
+    /**
+     * @param sequence
+     * @return A StreamDescriptor which is a copy of this one with the value of sequence changed to
+     *         the value passed in.  Recursively calls itself on the nestedStreams.
+     */
+    public StreamDescriptor copyWithSequence(int sequence)
```

Comment:
Given this method calls itself on child streams, I'd rather not inline it as it would involve a loop and a queue, or a something like that, and I think this looks cleaner.

Unless by inline it you just mean moving the method into MapFlattenedStreamReader, which I can definitely do.

Reply:
offline feedback: rename to withSequence

Reply:
@kevinwilfong : We agreed to have a private static method in `MapFlattenedStreamReader` to do this right? :)